core/src/main/scala/dagr/core/execsystem/TopLikeStatusReporter.scala

```diff
@@ -0,0 +1,214 @@
+    // get the total execution time, and total time since submission
+    val (executionTime: String, totalTime: String) = TaskExecutionInfo.getDurationSinceStartAndFormat(taskInfo)
+    // The state of execution
+    val graphNodeState = graphNodeStateFor(taskInfo.taskId).map(_.toString).getOrElse("NA")
+    List(
+      taskInfo.taskId.toString(),
+      taskInfo.task.name,
+      taskInfo.status.toString,
+      graphNodeState,
+      f"${taskInfo.resources.cores.value}%.2f",
+      taskInfo.resources.memory.prettyString,
+      timestampStringOrNA(taskInfo.submissionDate),
+      timestampStringOrNA(taskInfo.startDate),
+      timestampStringOrNA(taskInfo.endDate),
+      executionTime,
+      totalTime,
+      taskInfo.attemptIndex
+    ).map(_.toString)
+  }
+
+  /** Writes a delimited string of the status of all tasks managed to the console.
+    *
+    * @param print the method to use to write task status information, one line at a time.
+    * @param delimiter the delimiter between entries in a row.
+    */
+  def refresh(print: String => Unit, delimiter: String = "  "): Unit = {
+    val taskInfoMap: BiMap[Task, TaskExecutionInfo] = taskToInfoBiMapFor
+    val taskManagerResources = getTaskManagerResources
+    var numLinesLeft = Terminal.height
+
+    // Erase the screen, line-by-line.
+    for (i <- 0 until Terminal.height) {
+      print(CursorMovement.positionCursor(i, 0))
+      print(CursorMovement.eraseToEOL)
+    }
+    print(CursorMovement.positionCursor(0, 0))
+
+    //////////////////////////////////////////////////////////////////////////////////////////////
+    // Header row
+    //////////////////////////////////////////////////////////////////////////////////////////////
+    val header = new mutable.StringBuilder()
+
+    val unitTasks = taskInfoMap.filter { case (task, taskInfo) => task.isInstanceOf[UnitTask]}.map { case (task, taskInfo) => taskInfo }
+    val runningTasks: Map[UnitTask, ResourceSet] = runningTasksMap
+
+    val systemCoresPct: Double  = 100.0 * runningTasks.values.map(_.cores.value).sum / taskManagerResources.cores.value
+    val systemMemoryPct: Double = 100.0 * runningTasks.filterKeys(_.isInstanceOf[ProcessTask]).values.map(_.memory.value).sum / taskManagerResources.systemMemory.value
+    val jvmMemoryPct: Double    = 100.0 * runningTasks.filterKeys(_.isInstanceOf[InJvmTask]).values.map(_.memory.value).sum / taskManagerResources.jvmMemory.value
+
+    // NB: only considers unit tasks
+    val numRunning    = runningTasks.size
+    val numEligible   = readyTasksList.size
+    val numFailed     = unitTasks.count(taskInfo => TaskStatus.isTaskFailed(taskInfo.status))
+    val numDone       = unitTasks.count(taskInfo => TaskStatus.isTaskDone(taskInfo.status))
+    val numIneligible = unitTasks.count(taskInfo => taskInfo.status == TaskStatus.UNKNOWN)
+
+    header.append(s"Cores:         " + f"${taskManagerResources.cores.value}%6.1f" + ", " + f"$systemCoresPct%.1f" + "% Used\n")
```

Comment:
Seems odd on these lines to use a mix of `String.format`, `f""` and string concats with `+`.  Why not just go all the way and use a single `f""` or `String.format`per header line?